packages/sdk-effects/iframe-manager/src/lib/iframe-manager.effects.ts

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright (c) 2025 Ping Identity Corporation. All rights reserved.
+ *
+ * This software may be modified and distributed under the terms
+ * of the MIT license. See the LICENSE file for details.
+ */
+
+/* eslint-disable @typescript-eslint/no-empty-function */
+import { AuthorizeUrl } from '@forgerock/sdk-types';
+
+// Define specific options for the iframe request
+export interface IframeRequestOptions {
+  url: string;
+  timeout: number;
+  // Specify query parameters expected on success/error if needed,
+  // otherwise, parse all parameters found.
+  // Example: expectedParams?: string[];
+}
+
+type ResolvedParams<T> = Record<string, T>;
+
+type Noop = () => void;
+
+/**
+ * Initializes the Iframe Manager effect.
+ * This follows the functional effect pattern, returning an API
+ * within a closure. Configuration could be passed here if needed
+ * for default behaviors (e.g., default timeout), but currently,
+ * all config is per-request.
+ *
+ * @returns An object containing the API for managing iframe requests.
+ */
+export default function iframeManagerInit(/*config: OAuthConfig*/) {
+  /**
+   * Creates an iframe to make a background request to the specified URL,
+   * waits for a redirect, and resolves with the query parameters from the
+   * redirect URL.
+   *
+   * @param options - The options for the iframe request (URL, timeout).
+   * @returns A Promise that resolves with the query parameters from the redirect URL or rejects on timeout or error.
+   */
+  const getAuthCodeByIFrame = (options: {
+    url: AuthorizeUrl;
```

Comment:
Just wondering why we opted for a `url` param here instead of creating an authorize url based off some options passed in like the legacy sdk does? With the url param I would think a disadvantage for the end user is knowing how to construct a well formed authorize url.

Reply:
Justin wanted us to pass in the authorize url here rather than create it in this function

Reply:
Yes, this is in an effort to generalize this iframe function and "flatten" what these functions do for better composability.